File: backend/src/domain/job/job.py
from pydantic import BaseModel
from typing import Optional, Dict
from datetime import datetime
import uuid
from enum import Enum
from typing import Any
from pydantic import Field

class JobStatus(str, Enum):
    PENDING = 'Pending'
    RUNNING = 'Running'
    COMPLETED = 'Completed'
    FAILED = 'Failed'

class Job(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    worker_id: str
    name: str
    status: JobStatus = JobStatus.PENDING
    created_at: datetime = Field(default_factory=datetime.utcnow)
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    result: Optional[str] = None
    metadata: Dict[str, Any] = {}

class JobCreate(BaseModel):
    worker_id: str
    name: str
    metadata: Dict[str, Any] = {}

class JobUpdate(BaseModel):
    status: Optional[JobStatus] = None
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    result: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None

File: golang/devops_console/internal/domain/job/job.go
package job

import (
	"time"
)

type JobStatus string

const (
	JobStatusPending   JobStatus = "Pending"
	JobStatusRunning   JobStatus = "Running"
	JobStatusCompleted JobStatus = "Completed"
	JobStatusFailed    JobStatus = "Failed"
)

type Job struct {
	ID        string                 `json:"id"`
	WorkerID  string                 `json:"worker_id"`
	Name      string                 `json:"name"`
	Status    JobStatus              `json:"status"`
	CreatedAt time.Time              `json:"created_at"`
	StartedAt *time.Time             `json:"started_at,omitempty"`
	FinishedAt *time.Time            `json:"finished_at,omitempty"`
	Result    *string                `json:"result,omitempty"`
	Metadata  map[string]interface{} `json:"metadata"`
}

type JobCreate struct {
	WorkerID string                 `json:"worker_id"`
	Name     string                 `json:"name"`
	Metadata map[string]interface{} `json:"metadata"`
}

type JobUpdate struct {
	Status     *JobStatus              `json:"status,omitempty"`
	StartedAt  *time.Time              `json:"started_at,omitempty"`
	FinishedAt *time.Time              `json:"finished_at,omitempty"`
	Result     *string                 `json:"result,omitempty"`
	Metadata   map[string]interface{}  `json:"metadata,omitempty"`
}

type JobRepository interface {
	Create(job JobCreate) (*Job, error)
	GetAll() ([]*Job, error)
	GetByID(jobID string) (*Job, error)
	Update(jobID string, jobUpdate JobUpdate) (*Job, error)
	Delete(jobID string) error
	GetJobsByWorkerID(workerID string) ([]*Job, error)
}
